### PR TITLE
#cm 13

### PR DESCRIPTION
**Pull request by Chipman**
Added mixin for breakpoints. Added responcive design to menu.

### CODE REVIEW

File: app/scripts/modules/Menu.js [hunk 1]
```diff
@@ -3,16 +3,111 @@ var React = require('react');
...
+
+  function nullStripeMargin() {
+    menuButtonStripes[1].style.marginTop = '0px';
+    menuButtonStripes[1].style.marginBottom = '0px';
+  }
+  //make stripes on label closer together
+  label.onmouseout = nullStripeMargin;
+
+  //make stripes on label further apart
+  label.onmouseover = setStripeMargin;
```
Comment (tkharuk) [inline app/scripts/modules/Menu.js:82]: not appropriate. think of a different way.
e.g simple :hover with css animation

File: app/scripts/modules/Menu.js [hunk 2]
```diff
@@ -3,16 +3,111 @@ var React = require('react');
...
-          <a href="#speakers" className="menu__item">Speakers</a>
-          <a href="#shedule" className="menu__item">Shedule</a>
-          <a href="#location" className="menu__item">Location</a>
-          <a href="#registration" className="menu__item">Registration</a>
-          <a href="#partners" className="menu__item">Partners</a>
-      </nav>
+      <div className="menu-wrapper">
+        <div id="toggleWrapper">
+          <input type="checkbox" id="menuToggleBox" />
+          <label htmlFor="menuToggleBox" id="menuToggle">
```
Comment (tkharuk) [inline app/scripts/modules/Menu.js]: looking for "for" ?

File: app/scripts/modules/Menu.js [hunk 3]
```diff
@@ -3,16 +3,111 @@ var React = require('react');
...
-      <nav className="menu">
-          <a href="#overview" className="menu__item">Overview</a>
-          <a href="#speakers" className="menu__item">Speakers</a>
-          <a href="#shedule" className="menu__item">Shedule</a>
-          <a href="#location" className="menu__item">Location</a>
-          <a href="#registration" className="menu__item">Registration</a>
-          <a href="#partners" className="menu__item">Partners</a>
-      </nav>
+      <div className="menu-wrapper">
+        <div id="toggleWrapper">
```
Comment (tkharuk) [inline app/scripts/modules/Menu.js]: all ids should be prefixed like cm_toggleWrapper

File: app/scripts/modules/Menu.js [hunk 4]
```diff
@@ -3,16 +3,111 @@ var React = require('react');
...
 });
 
+//fix menu when scrolling os make static due to window.pageYOffset
+window.addEventListener('scroll', function() {
+  var menu = document.getElementById('menu');
+  var header = document.getElementById('header');
+  var overview = document.getElementById('overview');
+  var pageScroll = window.pageYOffset;
+  var menuStyle = menu.style;
+  var manuHeight = menu.offsetHeight;
```
Comment (tkharuk) [inline app/scripts/modules/Menu.js:47]: unused variable, with a typo

File: app/scripts/modules/Menu.js [hunk 5]
```diff
@@ -3,16 +3,111 @@ var React = require('react');
...
+        <div id="toggleWrapper">
+          <input type="checkbox" id="menuToggleBox" />
+          <label htmlFor="menuToggleBox" id="menuToggle">
+            <div className="menuToggle__stripe"></div>
+            <div className="menuToggle__stripe"></div>
+            <div className="menuToggle__stripe"></div>
+          </label>
+        </div>
+        <nav id="menuItems">
+            <a href="#overview" className="menu__item">Overview</a>
```
Comment (tkharuk) [inline app/scripts/modules/Menu.js]: menu__item class should belong to menu class

File: app/scripts/modules/Partners.js [hunk 6]
```diff
@@ -5,7 +5,7 @@ var React = require('react');
 var Partners = React.createClass({
```
Comment (tkharuk) [inline app/scripts/modules/Partners.js:5]: react file extension should be .jsx, not .js

File: app/styles/mixins.scss [hunk 7]
```diff
@@ -19,7 +20,25 @@
...
+/* speakers */
 @mixin social-button($image) {
 	background: $image;
 };
 
+
+/* media queries */
+@mixin bp($point) {
+  @if $point == medium {
+    @media (max-width: 768px) { @content; }
```
Comment (tkharuk) [inline app/styles/mixins.scss]: mixed tabs and spaces

File: app/styles/modules/_menu.scss [hunk 8]
```diff
@@ -4,20 +4,106 @@
 #menu {
 	@include container-ext;
 	background: $color__primary--dark;
+	z-index: 2;
+	@include bp(medium) {
+		position: fixed !important;
```
Comment (tkharuk) [inline app/styles/modules/_menu.scss]: do not use !important!

File: app/styles/modules/_menu.scss [hunk 9]
```diff
@@ -4,20 +4,106 @@
...
+		width: 100%;
+		top: 0px;
+	}
 }
 
-.menu {
+#toggleWrapper {
```
Comment (tkharuk) [inline app/styles/modules/_menu.scss]: do not use id for styling

File: app/styles/modules/_partners.scss [hunk 10]
```diff
@@ -12,45 +12,45 @@
...
 
 .partners__category-header {
-	color: $color__primary;
-	align-items: center;
-	font-size: 3em;
-	padding: 25px;
-	text-transform: capitalize;
-	-webkit-text-stroke-width: 1px;
-   -webkit-text-stroke-color: $color__primary--dark;
+    box-shadow: 0 34px 40px -33px rgba(26, 26, 26, 0.7);
```
Comment (tkharuk) [inline app/styles/modules/_partners.scss]: mixed tabs and spaces

File: app/scripts/modules/Menu.js (hunk 1 above)
Comment (TuxujPes) [inline app/scripts/modules/Menu.js]: line 75

File: app/scripts/modules/Menu.js [hunk 11]
```diff
@@ -3,16 +3,111 @@ var React = require('react');
...
+        checkBox.checked = false;
+        nullStripeMargin();
+      };
+    }
+  };
+
+  function toggle() {
+    if (!checkBox.checked) {
+      toggleDisplay(0 + 'px');
+      setStripeMargin('4.5px');
```
Comment (tkharuk) [inline app/scripts/modules/Menu.js:99]: do not use hardcoded, 'magic' numbers

File: app/scripts/modules/Menu.js [hunk 12]
```diff
@@ -3,16 +3,111 @@ var React = require('react');
...
+      label.onmouseout = setStripeMargin;
+      //add darken effect to screen
+      app.appendChild(darkenScreen);
+      darkenScreen.style.opacity = '1';
```
Comment (tkharuk) [inline app/scripts/modules/Menu.js]: use class '--modifier' instead of applying styles directly

File: app/scripts/modules/Menu.js [hunk 13]
```diff
@@ -3,16 +3,111 @@ var React = require('react');
...
+      darkenScreen.style.zIndex = '1';
+    } else {
+      toggleDisplay('-' + menuWidth + 'px');
+      label.onmouseout = nullStripeMargin;
+      darkenScreen.style.opacity = '0';
+      setTimeout(function() {
```
Comment (tkharuk) [inline app/scripts/modules/Menu.js]: use css animation instead of js timeout

File: app/scripts/modules/Menu.js [hunk 14]
```diff
@@ -3,16 +3,111 @@ var React = require('react');
...
+        darkenScreen.style.zIndex = '-1';
+      }, 300);
+    }
+  }
+
+  label.onclick = function() {
```
Comment (tkharuk) [inline app/scripts/modules/Menu.js:107]: use 'react' way of applying listeners

File: app/styles/modules/_menu.scss [hunk 15]
```diff
@@ -4,20 +4,106 @@
...
+	}
 }
 
 .menu__item{
+	display: block;
 	list-style:none;
 	font-size: $font__giant;
 	color: $color__heading--menu;
-	padding: 20px;
+	padding: $menu__item-padding--top-bottom $menu__item-padding--left-right;
```
Comment (tkharuk) [inline app/styles/modules/_menu.scss]: very odd naming for variable

File: app/styles/modules/_menu.scss [hunk 16]
```diff
@@ -4,20 +4,106 @@
...
+		padding: 6px 0;
+		height: $menu__toggle-button--size;
+		width: $menu__toggle-button--size;
+		margin-left: 10px;
+		transition: justify-content 0.2s;
+	}
+}
+
+.menuToggle__stripe {
+	height: $menuToggleBox__stripe--height;
```
Comment (tkharuk) [inline app/styles/modules/_menu.scss]: if you are trying to keep with BEM naming, --height does not seem to be a modifier

File: app/styles/modules/_partners.scss [hunk 17]
```diff
@@ -12,45 +12,45 @@
...
-
+	
 	img {
 		width: 100%;
+		transition: transform 0.2s;
 	}
 }
 
+.partners__item-link:hover img {
+	 transform: translatey(-3px);
```
Comment (tkharuk) [inline app/styles/modules/_partners.scss]: capitalize last 'Y' for readability

**Comment by Chipman** [comment]
i guess so.

File: app/scripts/modules/Menu.js [hunk 18]
```diff
@@ -3,16 +3,78 @@ var React = require('react');
...
+  }
+
+  //hide darken screen by clicking on it
+  darkenScreen.onclick = function() {
+    toggleDisplay();
+  }
+
+  //let on click on menu item togle visibility of menu
+  for (i = 0; i < menuItems.length; i++) {
+    menuItems[i].onclick = function() {
```
Comment (tkharuk) [inline app/scripts/modules/Menu.js]: should be onClick on element in render

**Pull request closed without merging**